src/components/generic-elements/download-button/download-button.js

```diff
@@ -1,5 +1,31 @@
 import './download-button.scss'
+import { createRef, useState } from 'react'
+import React from 'react'
+import {AuthenticatedHTTPClient} from '../../../utils/http-clients'
 
-export default (props) => (
-  <a className='download-button' href={props.url} target='_blank' rel='noopener noreferrer'>Download</a>
-)
+export function DownloadButton ({ url, format, filename }) {
+
+  const link = createRef()
+  const [downloadUrl, setDownloadUrl] = useState()
+  
+  React.useEffect(() => {
+    if (downloadUrl) {
+      link.current.click()
+    }
+  }, [downloadUrl])
+
+  const handleAction = async () => {
```

Comment:
This feels like the wrong place for this logic. I would try to get it out of here so that this component is more pure.

Reply:
Our dillema is that we're time constrained. We decided on a 1 min timeout for the signed url/token. That means that we need to make both calls within a minute, so it seems like the flow needs to be started by pressing the download button, and it shouldn't require a second click.

That being said, we both agree that this component feels weird. I'm interested in how we could make this better.

Reply:
The logic is now greatly simplified. It still does an async within the component but we feel a lot better about it now.